tests/unit/browser/test_qutehistory.py

```diff
@@ -0,0 +1,117 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+from PyQt5.QtCore import QUrl
+from qutebrowser.browser import history, qutescheme
+from qutebrowser.utils import objreg
+import datetime
+import pytest
+import tempfile
+
+
+class TestHistoryHandler:
+    """Test the qute://history endpoint."""
+
+    @pytest.fixture(autouse=True)
+    def fake_objects(self, fake_save_manager):
+        """ Create fake web-history with history for three different days """
+        try:
+            original_save_manager = objreg.get('save-manager')
+        except KeyError:
+            original_save_manager = None
+
+        try:
+            original_web_history = objreg.get('web-history')
+        except KeyError:
+            original_web_history = None
+
+        temp_dir = tempfile.TemporaryDirectory()
+        fake_web_history = history.WebHistory(temp_dir.name, 'fake-history')
+        objreg.register('web-history', fake_web_history, update=True)
+        objreg.register('save-manager', fake_save_manager, update=True)
+
+        # Add fake history items for different days
+        one_day = datetime.timedelta(days=1)
+        self.curr_date = datetime.datetime.today()
+        self.next_date = self.curr_date + one_day
+        self.prev_date = self.curr_date - one_day
+
+        today = history.Entry(atime=str(self.curr_date.timestamp()),
+            url=QUrl('www.today.com'), title='today')
+        tomorrow = history.Entry(atime=str(self.next_date.timestamp()),
+            url=QUrl('www.tomorrow.com'), title='tomorrow')
+        yesterday = history.Entry(atime=str(self.prev_date.timestamp()),
+            url=QUrl('www.yesterday.com'), title='yesterday')
+
+        web_history = objreg.get('web-history')
+        web_history._add_entry(today)
+        web_history._add_entry(tomorrow)
+        web_history._add_entry(yesterday)
+        web_history.save()
+
+        yield
+
+        if original_save_manager:
+            objreg.register('save-manager', original_save_manager, update=True)
+        else:
+            objreg.delete('save-manager')
+
+        if original_web_history:
+            objreg.register('web-history', original_web_history, update=True)
+        else:
+            objreg.delete('web-history')
+
+    def test_history_without_query(self):
+        """ Test qute://history shows today's history when it has no query """
+        _mimetype, data = qutescheme.qute_history(QUrl("qute://history"))
+        key = "<span class=\"date\">{}</span>".format(
+            datetime.date.today().strftime("%a, %d %B %Y"))
+        assert key in data
+
+    def test_history_with_bad_query(self):
+        """ Test qute://history shows today's history when given bad query """
```

Comment:
Those docstrings are a bit weird grammatically I think - but you can use `Ensure ...` everywhere instead.